### PR TITLE
feat: helper for transaction receipt

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution

New PR replacing PR #851 (due to creation of a new branch)

## Usage related changes

see PR #851

## Development related changes

see PR #851

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: src/types/provider/response.ts [hunk 1]
```diff
@@ -58,9 +64,45 @@ export type Block = {
...
-export type GetTransactionReceiptResponse = Receipt | PendingReceipt;
+export type RejectedTransactionReceiptResponse = {
+  status: `${TransactionStatus.REJECTED}`;
+  transaction_failure_reason: {
+    code: string;
+    error_message: string;
+  };
+};
+
+export type GetTransactionReceiptResponseWoHelper =
```
Comment (ivpavici) [inline src/types/provider/response.ts:95]: I suupose "wo" means "without", but what do you think of a different abbreviation, like: `GetTxReceiptResponseWithoutHelper` ?

File: src/utils/transactionReceipt.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * ```
+ */
+export class ReceiptTx implements TransactionReceiptUtilityInterface {
+  public readonly statusReceipt: TransactionReceiptStatus;
+
+  public readonly value: TransactionReceiptValue;
+
+  constructor(receipt: GetTransactionReceiptResponseWoHelper) {
+    [this.statusReceipt, this.value] = ReceiptTx.isSuccess(receipt)
+      ? ['success', receipt]
```
Comment (ivpavici) [inline src/utils/transactionReceipt.ts:41]: 'success' and rest of the statuses are repeated as magic strings also below on line 71 etc... maybe to extract them in a constant?

File: src/types/provider/response.ts (hunk 1 above)
Comment (PhilippeR26) [reply src/types/provider/response.ts:95]: Made

File: src/utils/transactionReceipt.ts (hunk 2 above)
Comment (PhilippeR26) [reply src/utils/transactionReceipt.ts:41]: The type TransactionReceiptStatus is listing these strings.
image
@penovicp Is it possible to create an enum based on the strings listed in a type?

**Review by tabaktoni — APPROVED** [review]
lgtm

**Review by tabaktoni — CHANGES_REQUESTED** [review]
Will need a free day to play with this to find a solution

**Comment by PhilippeR26** [comment]
> Will need a free day to play with this to find a solution

Or can't we merge quickly to avoid again conflicts, and see this polishing aspect as soon as you have a free day.

**Pull request merged**